src/middlewared/middlewared/plugins/failover.py

```diff
@@ -435,6 +435,22 @@ async def get_disks_local(self):
         except Exception:
             self.logger.error('Unhandled exception in get_disks_local', exc_info=True)
 
+    @private
+    def mismatch_nics(self):
+        """Determine if NICs match between both controllers."""
+        result = False
+        try:
+            local_nics = self.middleware.call_sync('interface.get_nic_names')
+            remote_nics = self.middleware.call_sync('failover.call_remote', 'interface.get_nic_names')
```

Comment:
You need to review the `raise_connect_error` argument to `failover.call_remote` instead of catching CallError here.